File: io/fib.rb
# Wrote this to help out with the io/fib.io script.
# Wanted to vet the idea after my io implementation
# wasn't giving me the right value.

def fibIndex(n)
	fibs = [1,1]
	fibs << fibs[-2] + fibs[-1] until n < fibs.length - 1
	fibs[n]
end

num = ARGV.pop.to_i || 5
puts "Fib at index #{num} is #{fibIndex(num)}"

File: haskell/fib.hs
module Main where
  fib :: Integer -> Integer
  fib 0 = 1
  fib 1 = 1
  fib x = fib (x - 1) + fib (x - 2)

  fibTuple :: (Integer, Integer, Integer) -> (Integer, Integer, Integer)
  fibTuple (x, y, 0) = (x, y, 0)
  fibTuple (x, y, index) = (y, x + y, index - 1)
